From 8f03fd6d507f35ef6b107f0ec176573160a53b90 Mon Sep 17 00:00:00 2001
From: =?UTF-8?q?Maxime=20M=C3=A9r=C3=A9?= <maxime.mere@foss.st.com>
Date: Fri, 30 Aug 2024 14:38:17 +0200
Subject: [PATCH 676/841] feat(stm32mp2): change PKH_OTP value for STM32MP21
MIME-Version: 1.0
Content-Type: text/plain; charset=UTF-8
Content-Transfer-Encoding: 8bit

Public key hash is stored in OTP 152 in STM32MP21 instead of OTP144

Upstream-Status: Inappropriate [ST downstream https://github.com/STMicroelectronics/arm-trusted-firmware.git e598c771574e49d48bf963ed3aeadc33b0e9c322 feat(stm32mp2): change PKH_OTP value for STM32MP21]
Signed-off-by: Maxime Méré <maxime.mere@foss.st.com>
Change-Id: Iba9a7d12963c9418b17d8bc92bb7f228e2ec868f
Reviewed-on: https://gerrit.st.com/c/mpu/oe/st/tf-a/+/401185
ACI: CIBUILD <MDG-smet-aci-builds@list.st.com>
Domain-Review: Yann GAUTIER <yann.gautier@foss.st.com>
ACI: CITOOLS <MDG-smet-aci-reviews@list.st.com>
Reviewed-by: Yann GAUTIER <yann.gautier@foss.st.com>
---
 plat/st/stm32mp2/stm32mp2_def.h | 4 ++++
 1 file changed, 4 insertions(+)

diff --git a/plat/st/stm32mp2/stm32mp2_def.h b/plat/st/stm32mp2/stm32mp2_def.h
index 5ca2d859e..ae6181a67 100644
--- a/plat/st/stm32mp2/stm32mp2_def.h
+++ b/plat/st/stm32mp2/stm32mp2_def.h
@@ -468,7 +468,11 @@ enum ddr_type {
 #define BOARD_ID_OTP			"board_id"
 #define UID_OTP				"uid_otp"
 #define LIFECYCLE2_OTP			"otp18"
+#if STM32MP21
+#define PKH_OTP				"otp152"
+#else
 #define PKH_OTP				"otp144"
+#endif
 #define ENCKEY_OTP			"otp260"
 
 /* OTP mask */
-- 
2.47.2

